File: Dia10/d10p1/src/main.py
with open('../../input.txt') as f:
	lines =  f.readlines()
	count = 0
	for line in lines:
		stack = []
		for c in line.strip():
			if c in '([{<':
				stack.append(c)
			else:
				x = stack.pop()
				if (x == '(' and c != ')') or (x != '(' and ord(c) - ord(x) != 2):
					if c == ')':
						count += 3
					elif c == ']':
						count += 57
					elif c == '}':
						count += 1197
					elif c == '>':
						count += 25137
					break
	print(count)

File: Dia10/d10p1/src/main.rs
fn main() {
    let sum = include_str!("../../input.txt")
        .lines()
        .fold(0, |count, line| {
            let mut stack = vec![];
            let mut last = None;
            for c in line.chars() {
                if let '(' | '[' | '{' | '<' = c {
                    stack.push(c)
                } else if !matches!(
                    (stack.pop().unwrap(), c),
                    ('(', ')') | ('[', ']') | ('{', '}') | ('<', '>')
                ) {
                    last = Some(c);
                    break;
                }
            }
            // println!("L {} :: {:?}", line, last);
            count
                + match last {
                    Some(')') => 3,
                    Some(']') => 57,
                    Some('}') => 1197,
                    Some('>') => 25137,
                    _ => 0,
                }
        });

    println!(">> {}", sum);
}
